cmd/plugins/cdi-device-injector/cdi-device-injector.go

```diff
@@ -0,0 +1,201 @@
+// Copyright The NRI Plugins Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+	"tags.cncf.io/container-device-interface/pkg/cdi"
+	"tags.cncf.io/container-device-interface/pkg/parser"
+
+	"github.com/containerd/nri/pkg/api"
+	"github.com/containerd/nri/pkg/stub"
+)
+
+const (
+	cdiDeviceKey = "cdi.nri.io"
+)
+
+var (
+	log     *logrus.Logger
+	verbose bool
+)
+
+// our injector plugin
+type plugin struct {
+	stub                    stub.Stub
+	allowedCDIDevicePattern string
+	cdiCache                *cdiCache
+}
+
+// CreateContainer handles container creation requests.
+func (p *plugin) CreateContainer(ctx context.Context, pod *api.PodSandbox, container *api.Container) (_ *api.ContainerAdjustment, _ []*api.ContainerUpdate, err error) {
+	defer func() {
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+	name := containerName(pod, container)
+
+	if verbose {
+		dump("CreateContainer", "pod", pod, "container", container)
+	} else {
+		log.Infof("CreateContainer %s", name)
+	}
+
+	if p.allowedCDIDevicePattern == "" {
+		return nil, nil, nil
+	}
+
+	cdiDevices, err := parseCdiDevices(pod.Annotations, container.Name)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to parse CDI Device annotations: %w", err)
+	}
+
+	if len(cdiDevices) == 0 {
+		return nil, nil, nil
+	}
+
+	var allowedCDIDevices []string
+	for _, cdiDevice := range cdiDevices {
+		match, _ := filepath.Match(p.allowedCDIDevicePattern, cdiDevice)
```

Comment:
I think it would be better to use `path.Match()` for globbing instead of `filepath.Match()`.